process_csv.php

```diff
@@ -0,0 +1,218 @@
+                $manifest_path = $tempdir . '/manifest.csv';
+                if (file_exists($manifest_path)) {
+                    $missing_files = csv_client_helper::check_manifest_and_files($manifest_path, $tempdir);
+                    if (!empty($missing_files['missing_files']) && !empty($missing_files['invalid_headers'])) {
+                        echo $OUTPUT->header();
+                        csv_client_helper::display_missing_and_invalid_files($missing_files);
+                        echo $OUTPUT->footer();
+                        exit;
+                    } 
+
+                    $datatype = csv_client_helper::validate_csv_data_types($tempdir);
+                    if (!$datatype['is_valid']) {
+                        echo $OUTPUT->header();
+                        csv_client_helper::display_validation_errors($datatype);
+                        echo $OUTPUT->footer();
+                        exit;
+                    }
+
+                    $csv_data = csv_client_helper::extract_csvs_to_arrays($tempdir);
+                    $orgs = $csv_data['orgs'] ?? [];
+                    // Prepare organization options
+                    $orgoptions = [];
+                    foreach ($orgs as $org) {
+                        $orgoptions[$org['sourcedId']] = $org['name'];
+                    }
+                    if (count($orgoptions) == 1) {
+                        // Only one organization, skip the selection form
+                        $selected_org_sourcedId = array_key_first($orgoptions);
+                        // Proceed to process the selected organization
+                        process_selected_organization($selected_org_sourcedId, $tempdir, $csv_data);
+                        exit;
+                    } else {
+                        // Display the organization selection form
+                        $SESSION->oneroster_csv['orgoptions'] = $orgoptions;
+                        $orgform = new oneroster_org_selection_form(null, ['orgoptions' => $orgoptions, 'tempdir' => $tempdir]);
+                        echo $OUTPUT->header();
+                        $orgform->display();
+                        echo $OUTPUT->footer();
+                        exit;
+                    } 
+                } else {
+                echo $OUTPUT->header();
+                echo get_string('missing_csv_files', 'enrol_oneroster') . ' <br>';
+                echo $OUTPUT->footer();
+                exit;
+                }
+            } 
+            else {
+                echo $OUTPUT->header();
+                echo get_string('failed_to_open_zip_file', 'enrol_oneroster') . ' <br>';
+                echo $OUTPUT->footer();
+                exit;
+            }
+        } 
+        else {
+            echo $OUTPUT->header();
+            echo get_string('failed_upload_zip_file', 'enrol_oneroster') . ' <br>';
+            echo $OUTPUT->footer();
+            exit;
+        }
+    } 
+    else {
```

Comment:
Update this to be a series of guard clauses that redirect back to the start with a message